core/src/hir/types.rs

```diff
@@ -11,7 +11,7 @@ pub enum ReturnableType {
 /// Type that can only be used as an output.
 pub enum OutType {
     Primitive(PrimitiveType),
-    Opaque(paths::OutOpaque),
+    Opaque(paths::ReturnedOpaque),
```

Comment:
this can still be split into OpaqueRef(OpaqueRef) and OpaqueOwned(OpaqueOwned)

Reply:
I think it's better to have one type that contains the `Ownership` enum because otherwise it gets annoying to write code that is shared between most of them except for one small piece of logic that changes based on whether or not its owned.

Reply:
Alright. I'm under the impression it's not super shared but we can see!